### PR TITLE
list_files: Don't crash for empty directories

### PR DESCRIPTION
**Pull request by ekmartin**
Also adds an include_hidden option to list_files, with default to False to prevent breaking backwards compatibility.

### CODE REVIEW

File: docker/manager.py [hunk 1]
```diff
@@ -183,17 +185,19 @@ def list_files(self, path):
...
 
         if not result.succeeded:
             if errors.FILE_NOT_FOUND_PREDICATE in result.err:
                 raise errors.DockerFileNotFoundError(path)
 
             raise errors.DockerWrapperBaseError(result.err)
 
-        return result.out.strip().split('\n')
+        out = result.out.strip()
+        return sorted(out.split('\n')) if out else []
```
Comment (ekmartin) [inline docker/manager.py:200]: `find` produces a different ordering than what we had previously, I think sorting it before return results in the behavior we've had before.

File: tests/test_manager.py [hunk 2]
```diff
@@ -261,7 +281,7 @@ def test_combine_output(self):
         result = self.docker.run('ls does-not-exist')
         self.assertEqual(result.err, '')
         self.assertEqual(result.out,
-                         'ls: cannot access does-not-exist: No such file or directory\n')
+                         "ls: cannot access 'does-not-exist': No such file or directory\n")
```
Comment (ekmartin) [inline tests/test_manager.py:284]: This test failed because of the missing quotes.

Comment (ekmartin) [inline tests/test_manager.py:284]: Might be because the default for ubuntu has changed from 14.04 to 16.04.

**Comment by relekang** [comment]
:shipit:

**Pull request merged**